cmake/OpenCVCompilerOptions.cmake

```diff
@@ -29,6 +33,12 @@ if(ENABLE_CCACHE AND NOT OPENCV_COMPILER_IS_CCACHE AND NOT CMAKE_GENERATOR MATCH
         set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${__OLD_RULE_LAUNCH_COMPILE}")
       endif()
     endif()
+    if((NOT ${CCACHE_VERSION_MAJOR} LESS 3) AND (NOT ${CCACHE_VERSION_MINOR} LESS 4))
```

Comment:
CMake has [builtin version comparison support](https://cmake.org/cmake/help/v3.14/command/if.html?highlight=VERSION_): `if(<variable|string> VERSION_LESS <variable|string>)`

So `CCACHE_VERSION_MAJOR|MINOR` extraction code is not needed.

Reply:
Yes, that's much better ! Thank you !